plugins/org.eclipse.cbi.p2repo.aggregator.engine.maven/src/org/eclipse/cbi/p2repo/aggregator/engine/maven/InstallableUnitMapping.java

```diff
@@ -117,15 +119,14 @@ public enum Type {
 
 	private boolean hasSources;
 
-	public InstallableUnitMapping(Contribution contribution, IInstallableUnit iu) {
-		this(contribution, iu, Collections.<MavenMapping> emptyList());
-	}
+	private Collection<InstallableUnitMapping> allIUMapping;
 
-	public InstallableUnitMapping(Contribution contribution, IInstallableUnit iu, List<MavenMapping> mappings) {
+	public InstallableUnitMapping(Contribution contribution, IInstallableUnit iu, List<MavenMapping> mappings,
+			Collection<InstallableUnitMapping> allIUMapping) {
 		this.contribution = contribution;
+		this.allIUMapping = allIUMapping;
 		this.mappings = new ArrayList<MavenMapping>(mappings.size() + 1);
 		this.mappings.addAll(mappings);
-		this.mappings.add(MavenMapping.DEFAULT_MAPPING);
```

Comment:
If we keep this rule, then it means that all artifacts will be remapped even if they have p2 metadata. So it basically invalidates the capability to use another strategy for resolution.
What we can try instead if to use this rule in the engine instead of failing by default. So the flow would be first try a defined mapping rule, if none found use p2 metadata, if none found, use default mapping. WDYT?

Reply:
I've appended a commit to this PR to use MavenMapping.DEFAULT_MAPPING when nothing better can be found.

Reply:
For Platform aggregator, I don't think we do need such a fine grain as such a template rule; good sensible default are enough. However, if you believe this is really helpful to other builds, I won't refrain you from doing it as long as it serves Platform well. I'm just not sure whether it's worth it in general, but I'm focused on Platform at the moment and may miss some context to make such a rule valuable.

Reply:
Yes, it seems to me that often the p2 maven metadata is wrong and of course the default rule is no doubt wrong just as often.  Better would have been to fail when no rule kicked it, but that's history...

My top priority is to help the platform be able to publish snapshots that are of reliable quality as soon as possible (without breaking anyone else).